src/main/java/com/mewna/catnip/extension/Extension.java

```diff
@@ -154,61 +126,14 @@ default Extension injectOptions(@Nonnull final Function<CatnipOptions, CatnipOpt
     }
     
     /**
-     * Add a consumer for the specified event type.
-     * <p>
-     * This method behaves similarly to {@link Catnip#on(EventType)},
-     * but the event listeners are unregistered when the extension unloads
-     *
-     * @param type The type of event to listen on.
-     * @param <T>  The object type of event being listened on.
-     *
-     * @return The vert.x message consumer.
-     */
-    <T> MessageConsumer<T> on(@Nonnull EventType<T> type);
-    
-    /**
-     * Add a consumer for the specified event type with the given handler
-     * callback.
-     * <p>
-     * This method behaves similarly to {@link Catnip#on(EventType, Consumer)},
-     * but the event listeners are unregistered when the extension unloads
-     *
-     * @param type    The type of event to listen on.
-     * @param handler The handler for the event object.
-     * @param <T>     The object type of event being listened on.
-     *
-     * @return The vert.x message consumer.
+     * Callback called once the Extension has been loaded by the {@link ExtensionManager}.
      */
-    <T> MessageConsumer<T> on(@Nonnull EventType<T> type, @Nonnull Consumer<T> handler);
-    
-    /**
-     * Add a consumer for the specified event type.
-     * <p>
-     * This method behaves similarly to {@link Catnip#on(DoubleEventType)},
-     * but the event listeners are unregistered when the extension unloads
-     *
-     * @param type The type of event to listen on.
-     * @param <T>  The first type of event being listened on.
-     * @param <E>  The second type of event being listened on.
-     *
-     * @return The vert.x message consumer.
-     */
-    <T, E> MessageConsumer<Pair<T, E>> on(@Nonnull DoubleEventType<T, E> type);
+    default void onLoaded() {
```

Comment:
Also, not sure if it would make sense to make this return a `Completable` instead of `void`

Reply:
`Scheduler#scheduleDirect` receives a `Runnable` so not even sure how to handle a `Completable` return in this case.

Reply:
That's fair; I wasn't sure if it would even make sense to do in the first place. Was just a thought since some extensions may need async. startup, and it would help with running them all in serial.

Reply:
Uhh, in this PR, all extensions are loaded asynchronously from one another. As soon as you load the Extension on the Manager it sets the Extension up without awaiting any previous extension mid-loading.

Reply:
Well part of my concern is something like
1. start bot
2. loading extension that caches stuff in redis or whatever
3. bot starts recving events and needing to cache before extension is connected to redis

Reply:
I can solve it by making load/unloadExtension return a Completable, and the user chooses how to deal with it?

Reply:
fixed by making it synchronous (again)